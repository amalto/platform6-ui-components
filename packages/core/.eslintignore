dist/
loader/

src/components.d.ts