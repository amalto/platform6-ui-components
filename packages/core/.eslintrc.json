{
  "extends": ["../../.eslintrc.json", "plugin:@stencil/recommended", "plugin:jest/recommended", "plugin:jsx-a11y/recommended"],
  "parserOptions": {
    "project": ["./tsconfig.json", "./tsconfig.lint.json"],
    "ecmaFeatures": {
      "jsx": true // Allows for the parsing of JSX
    }
  },
  "plugins": ["jest"],

  "rules": {
    "jsx-a11y/label-has-associated-control": [
      2,
      {
        "assert": "either"
      }
    ],
    "react/jsx-props-no-spreading": "off",
    "react/no-unknown-property": [2, { "ignore": ["class"] }]
  },
  "settings": {
    "react": {
      "pragma": "h"
    }
  }
}
