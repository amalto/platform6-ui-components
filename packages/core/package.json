{
  "name": "@platform6/components",
  "version": "0.0.1",
  "description": "Platform 6 UI components",
  "repository": {
    "type": "git",
    "url": "https://github.com/amalto/platform6-ui-components"
  },
  "license": "MIT",
  "main": "dist/index.cjs.js",
  "module": "dist/custom-elements/index.js",
  "collection": "dist/collection/collection-manifest.json",
  "collection:main": "dist/collection/index.js",
  "es2015": "dist/esm/index.mjs",
  "es2017": "dist/esm/index.mjs",
  "unpkg": "dist/stencil-component/stencil-component.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/",
    "loader/"
  ],
  "scripts": {
    "build": "stencil build --docs",
    "format": "prettier --write ./",
    "postformat": "npm run lint.fix",
    "generate": "stencil generate",
    "lint": "npm run lint.ts && npm run lint.sass",
    "lint.fix": "npm run lint.ts.fix && npm run lint.sass.fix",
    "lint.sass": "stylelint 'src/**/*.scss'",
    "lint.sass.fix": "npm run lint.sass -- --fix",
    "lint.ts": "eslint --cache --ext .js,.jsx,.ts,.tsx ./",
    "lint.ts.fix": "npm run lint.ts -- --fix",
    "start": "stencil build --docs --dev --watch --serve",
    "test": "stencil test --spec --e2e",
    "test.watch": "stencil test --spec --e2e --watchAll",
    "storybook": "start-storybook -p 6006 -s ./dist --ci",
    "build-storybook": "build-storybook"
  },
  "dependencies": {
    "@creativebulma/bulma-tooltip": "^1.2.0",
    "@fortawesome/fontawesome-svg-core": "^1.2.32",
    "@fortawesome/free-brands-svg-icons": "^5.15.1",
    "@fortawesome/free-regular-svg-icons": "^5.15.1",
    "@fortawesome/free-solid-svg-icons": "^5.15.1",
    "@stencil/core": "2.2.0",
    "bulma": "^0.9.1",
    "bulma-calendar": "^6.0.9",
    "choices.js": "9.0.1",
    "lit-html": "^1.3.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.3",
    "@stencil/eslint-plugin": "^0.3.1",
    "@stencil/postcss": "^1.0.1",
    "@stencil/react-output-target": "0.0.9",
    "@stencil/sass": "^1.3.2",
    "@storybook/addon-a11y": "^6.0.28",
    "@storybook/addon-actions": "^6.0.28",
    "@storybook/addon-essentials": "^6.0.28",
    "@storybook/addon-links": "^6.0.28",
    "@storybook/html": "^6.0.28",
    "@storybook/preset-scss": "^1.0.3",
    "@storybook/theming": "^6.0.28",
    "@storybook/web-components": "^6.0.28",
    "@types/jest": "^26.0.15",
    "@types/js-beautify": "^1.11.0",
    "@typescript-eslint/eslint-plugin": "^4.6.1",
    "@typescript-eslint/parser": "^4.6.1",
    "autoprefixer": "^9.8.6",
    "babel-loader": "^8.1.0",
    "css-loader": "^5.0.0",
    "eslint-plugin-jest": "^24.1.0",
    "jest": "^26.6.3",
    "jest-cli": "^26.6.3",
    "js-beautify": "^1.13.0",
    "prettier": "^2.1.2",
    "puppeteer": "^5.4.1",
    "react-is": "^17.0.1",
    "sass-loader": "^10.0.5",
    "style-loader": "^2.0.0",
    "typescript": "4.0.5"
  }
}
