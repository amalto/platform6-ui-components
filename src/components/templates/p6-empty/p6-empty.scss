@import '@fortawesome/fontawesome-svg-core/styles';

:host {
  height: 100%;
  padding: 0;
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  margin: auto;
}
