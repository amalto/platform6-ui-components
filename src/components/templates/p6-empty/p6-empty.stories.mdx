import {
  Story,
  Preview,
  Meta,
  Description,
} from "@storybook/addon-docs/blocks";

import Readme from "./readme.md";

import {
  DefaultStory,
  CenteredStory,
  MessageStory,
  IconStory,
  ImageAndMessageStory,
  LanguageStory,
} from "./p6-empty.stories.ts";

<Meta title="Templates/Empty" component="p6-empty" />

# Empty

`p6-empty` provides an empty message

<Preview>
  <Story
    name="Default"
    height="84px"
    parameters={DefaultStory.parameters}
    args={DefaultStory.args}
    argTypes={DefaultStory.argTypes}
  >
    {DefaultStory}
  </Story>
</Preview>

<Preview>
  <Story
    name="Centered"
    height="300px"
    parameters={CenteredStory.parameters}
    args={CenteredStory.args}
    argTypes={CenteredStory.argTypes}
  >
    {CenteredStory}
  </Story>
</Preview>

<Preview>
  <Story
    name="With message"
    height="300px"
    parameters={MessageStory.parameters}
    args={MessageStory.args}
    argTypes={MessageStory.argTypes}
  >
    {MessageStory}
  </Story>
</Preview>

<Preview>
  <Story
    name="With icon"
    height="300px"
    parameters={IconStory.parameters}
    args={IconStory.args}
    argTypes={IconStory.argTypes}
  >
    {IconStory}
  </Story>
</Preview>

<Preview>
  <Story
    name="With image and message"
    height="300px"
    parameters={ImageAndMessageStory.parameters}
    args={ImageAndMessageStory.args}
    argTypes={ImageAndMessageStory.argTypes}
  >
    {ImageAndMessageStory}
  </Story>
</Preview>

<Preview>
  <Story
    name="Language"
    height="300px"
    parameters={LanguageStory.parameters}
    args={LanguageStory.args}
    argTypes={LanguageStory.argTypes}
  >
    {LanguageStory}
  </Story>
</Preview>

<Description markdown={Readme} />
