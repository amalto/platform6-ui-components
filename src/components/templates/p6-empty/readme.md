# p6-empty

<!-- Auto Generated Below -->


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
