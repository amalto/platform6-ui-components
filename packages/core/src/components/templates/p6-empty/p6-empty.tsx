import { icon } from '@fortawesome/fontawesome-svg-core';
import { Component, Element, h, Host, JSX } from '@stencil/core';
import { getL10n, L10n } from '../../../utils/translations';

@Component({
  tag: 'p6-empty',
  styleUrl: 'p6-empty.scss',
  shadow: true,
  assetsDirs: ['locales'],
})
export class P6Empty {
  @Element() element!: HTMLP6EmptyElement;

  private l10n: L10n | undefined;

  async componentWillLoad(): Promise<void> {
    this.l10n = await getL10n(this.element);
  }

  render(): JSX.Element {
    const img = icon({ prefix: 'far', iconName: 'folder-open' }, { classes: ['fa-3x'] });

    return (
      <Host>
        <slot name="image">
          <span class="icon is-large" innerHTML={img.html[0]} />
        </slot>
        <slot>{this.l10n?.emptyMessage}</slot>
      </Host>
    );
  }
}
