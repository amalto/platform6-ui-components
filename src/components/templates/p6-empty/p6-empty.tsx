import { icon } from "@fortawesome/fontawesome-svg-core";
import { Component, h, Host } from "@stencil/core";

@Component({
  tag: "p6-empty",
  styleUrl: "p6-empty.scss",
  shadow: true,
})
export class P6Empty {
  render(): JSX.Element {
    const img = icon(
      { prefix: "far", iconName: "folder-open" },
      { classes: ["fa-3x"] }
    );

    return (
      <Host>
        <slot name="image">
          <span class="icon is-large" innerHTML={img.html[0]} />
        </slot>
        <slot>Sorry, nothing to display</slot>
      </Host>
    );
  }
}
