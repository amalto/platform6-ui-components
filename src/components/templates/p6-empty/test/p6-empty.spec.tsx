import { library } from "@fortawesome/fontawesome-svg-core";
import { faFolderOpen } from "@fortawesome/free-regular-svg-icons";
import { newSpecPage } from "@stencil/core/testing";
import { P6Empty } from "../p6-empty";

library.add(faFolderOpen);

describe("p6-empty", () => {
  it("renders", async () => {
    const page = await newSpecPage({
      components: [P6Empty],
      html: `<p6-empty></p6-empty>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("with message", async () => {
    const page = await newSpecPage({
      components: [P6Empty],
      html: `<p6-empty>No data!</p6-empty>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("with image", async () => {
    const page = await newSpecPage({
      components: [P6Empty],
      html: `<p6-empty><p6-icon slot='image' name='home'/></p6-empty>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("with image and message", async () => {
    const page = await newSpecPage({
      components: [P6Empty],
      html: `<p6-empty><p6-icon slot='image' name='home'/>No data!</p6-empty>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
