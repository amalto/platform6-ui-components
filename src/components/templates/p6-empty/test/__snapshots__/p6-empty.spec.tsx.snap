// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p6-empty renders 1`] = `
<p6-empty>
  <mock:shadow-root>
    <slot name="image">
      <span class="icon is-large">
        <svg aria-hidden="true" class="fa-3x fa-folder-open fa-w-18 svg-inline--fa" data-icon="folder-open" data-prefix="far" focusable="false" role="img" viewBox="0 0 576 512" xmlns="http://www.w3.org/2000/svg">
          <path d="M527.9 224H480v-48c0-26.5-21.5-48-48-48H272l-64-64H48C21.5 64 0 85.5 0 112v288c0 26.5 21.5 48 48 48h400c16.5 0 31.9-8.5 40.7-22.6l79.9-128c20-31.9-3-73.4-40.7-73.4zM48 118c0-3.3 2.7-6 6-6h134.1l64 64H426c3.3 0 6 2.7 6 6v42H152c-16.8 0-32.4 8.8-41.1 23.2L48 351.4zm400 282H72l77.2-128H528z" fill="currentColor"></path>
        </svg>
      </span>
    </slot>
    <slot>
      Sorry, nothing to display!
    </slot>
  </mock:shadow-root>
</p6-empty>
`;

exports[`p6-empty translation Should translate en 1`] = `
<p6-empty>
  <mock:shadow-root>
    <slot name="image">
      <span class="icon is-large">
        <svg aria-hidden="true" class="fa-3x fa-folder-open fa-w-18 svg-inline--fa" data-icon="folder-open" data-prefix="far" focusable="false" role="img" viewBox="0 0 576 512" xmlns="http://www.w3.org/2000/svg">
          <path d="M527.9 224H480v-48c0-26.5-21.5-48-48-48H272l-64-64H48C21.5 64 0 85.5 0 112v288c0 26.5 21.5 48 48 48h400c16.5 0 31.9-8.5 40.7-22.6l79.9-128c20-31.9-3-73.4-40.7-73.4zM48 118c0-3.3 2.7-6 6-6h134.1l64 64H426c3.3 0 6 2.7 6 6v42H152c-16.8 0-32.4 8.8-41.1 23.2L48 351.4zm400 282H72l77.2-128H528z" fill="currentColor"></path>
        </svg>
      </span>
    </slot>
    <slot>
      Sorry, nothing to display!
    </slot>
  </mock:shadow-root>
</p6-empty>
`;

exports[`p6-empty translation Should translate fr 1`] = `
<p6-empty>
  <mock:shadow-root>
    <slot name="image">
      <span class="icon is-large">
        <svg aria-hidden="true" class="fa-3x fa-folder-open fa-w-18 svg-inline--fa" data-icon="folder-open" data-prefix="far" focusable="false" role="img" viewBox="0 0 576 512" xmlns="http://www.w3.org/2000/svg">
          <path d="M527.9 224H480v-48c0-26.5-21.5-48-48-48H272l-64-64H48C21.5 64 0 85.5 0 112v288c0 26.5 21.5 48 48 48h400c16.5 0 31.9-8.5 40.7-22.6l79.9-128c20-31.9-3-73.4-40.7-73.4zM48 118c0-3.3 2.7-6 6-6h134.1l64 64H426c3.3 0 6 2.7 6 6v42H152c-16.8 0-32.4 8.8-41.1 23.2L48 351.4zm400 282H72l77.2-128H528z" fill="currentColor"></path>
        </svg>
      </span>
    </slot>
    <slot>
      Aucune information Ã  afficher
    </slot>
  </mock:shadow-root>
</p6-empty>
`;

exports[`p6-empty with image 1`] = `
<p6-empty>
  <mock:shadow-root>
    <slot name="image">
      <span class="icon is-large">
        <svg aria-hidden="true" class="fa-3x fa-folder-open fa-w-18 svg-inline--fa" data-icon="folder-open" data-prefix="far" focusable="false" role="img" viewBox="0 0 576 512" xmlns="http://www.w3.org/2000/svg">
          <path d="M527.9 224H480v-48c0-26.5-21.5-48-48-48H272l-64-64H48C21.5 64 0 85.5 0 112v288c0 26.5 21.5 48 48 48h400c16.5 0 31.9-8.5 40.7-22.6l79.9-128c20-31.9-3-73.4-40.7-73.4zM48 118c0-3.3 2.7-6 6-6h134.1l64 64H426c3.3 0 6 2.7 6 6v42H152c-16.8 0-32.4 8.8-41.1 23.2L48 351.4zm400 282H72l77.2-128H528z" fill="currentColor"></path>
        </svg>
      </span>
    </slot>
    <slot>
      Sorry, nothing to display!
    </slot>
  </mock:shadow-root>
  <p6-icon name="home" slot="image"></p6-icon>
</p6-empty>
`;

exports[`p6-empty with image and message 1`] = `
<p6-empty>
  <mock:shadow-root>
    <slot name="image">
      <span class="icon is-large">
        <svg aria-hidden="true" class="fa-3x fa-folder-open fa-w-18 svg-inline--fa" data-icon="folder-open" data-prefix="far" focusable="false" role="img" viewBox="0 0 576 512" xmlns="http://www.w3.org/2000/svg">
          <path d="M527.9 224H480v-48c0-26.5-21.5-48-48-48H272l-64-64H48C21.5 64 0 85.5 0 112v288c0 26.5 21.5 48 48 48h400c16.5 0 31.9-8.5 40.7-22.6l79.9-128c20-31.9-3-73.4-40.7-73.4zM48 118c0-3.3 2.7-6 6-6h134.1l64 64H426c3.3 0 6 2.7 6 6v42H152c-16.8 0-32.4 8.8-41.1 23.2L48 351.4zm400 282H72l77.2-128H528z" fill="currentColor"></path>
        </svg>
      </span>
    </slot>
    <slot>
      Sorry, nothing to display!
    </slot>
  </mock:shadow-root>
  <p6-icon name="home" slot="image">
    No data!
  </p6-icon>
</p6-empty>
`;

exports[`p6-empty with message 1`] = `
<p6-empty>
  <mock:shadow-root>
    <slot name="image">
      <span class="icon is-large">
        <svg aria-hidden="true" class="fa-3x fa-folder-open fa-w-18 svg-inline--fa" data-icon="folder-open" data-prefix="far" focusable="false" role="img" viewBox="0 0 576 512" xmlns="http://www.w3.org/2000/svg">
          <path d="M527.9 224H480v-48c0-26.5-21.5-48-48-48H272l-64-64H48C21.5 64 0 85.5 0 112v288c0 26.5 21.5 48 48 48h400c16.5 0 31.9-8.5 40.7-22.6l79.9-128c20-31.9-3-73.4-40.7-73.4zM48 118c0-3.3 2.7-6 6-6h134.1l64 64H426c3.3 0 6 2.7 6 6v42H152c-16.8 0-32.4 8.8-41.1 23.2L48 351.4zm400 282H72l77.2-128H528z" fill="currentColor"></path>
        </svg>
      </span>
    </slot>
    <slot>
      Sorry, nothing to display!
    </slot>
  </mock:shadow-root>
  No data!
</p6-empty>
`;
